lib/web_ui/lib/src/engine/canvaskit/image.dart

```diff
@@ -15,24 +15,33 @@ void skiaInstantiateImageCodec(Uint8List list, Callback<ui.Codec> callback,
   callback(codec);
 }
 
-/// Instantiates a [ui.Codec] backed by an `SkAnimatedImage` from Skia after requesting from URI.
-void skiaInstantiateWebImageCodec(String src, Callback<ui.Codec> callback,
-    WebOnlyImageCodecChunkCallback? chunkCallback) {
+/// Instantiates a [ui.Codec] backed by an `SkAnimatedImage` from Skia after
+/// requesting from URI.
+Future<ui.Codec?> skiaInstantiateWebImageCodec(
+    String src, WebOnlyImageCodecChunkCallback? chunkCallback) {
+  Completer<ui.Codec?> completer = Completer<ui.Codec?>();
   chunkCallback?.call(0, 100);
   //TODO: Switch to using MakeImageFromCanvasImageSource when animated images are supported.
   html.HttpRequest.request(
     src,
     responseType: "arraybuffer",
   ).then((html.HttpRequest response) {
+    if (response.status != 200) {
+      completer.completeError(
+          'Network image request failed with status: ${response.status}');
```

Comment:
Given the amount of confusion this is causing, I think it's worth elaborating on what might be going on here. Alternatively, we could have a wiki page that explains CORS and link to it from here.

Reply:
CORS exceptions are caught in the onError handler below. This catches "successful" requests which aren't status 200. For CORS exceptions an error is dumped to the console explaining that the image to be downloaded didn't have CORS headers set.